### PR TITLE
kie-issues#1549: kn-workflow-plugin check for the presence of an image in the local Docker image does not cover all cases

### LINKED ISSUE #1549: kn-workflow-plugin check for the presence of an image in the local Docker image does not cover all cases
We should add better checks. because --filter=reference="*/image:main" doesn't work for all cases

### PR DESCRIPTION
**Pull request by treblereel**
Closes https://github.com/apache/incubator-kie-issues/issues/1549

### CODE REVIEW

File: packages/kn-plugin-workflow/pkg/common/containers.go [hunk 1]
```diff
@@ -51,6 +51,10 @@ type DockerLogMessage struct {
 	ID     string `json:"id,omitempty"`
 }
 
+type DockerClient interface {
+	ImageList(ctx context.Context, options types.ImageListOptions) ([]types.ImageSummary, error)
```
Comment (ricardozanini) [inline packages/kn-plugin-workflow/pkg/common/containers.go:58]: Does it work for podman too? 🤔

Comment (treblereel) [reply packages/kn-plugin-workflow/pkg/common/containers.go:58]: @ricardozanini you can take a look at `func RunContainerCommand` at containers.go,  only docker uses it.

I plan to refactor it a little in the future, this func can be improved

Comment (ricardozanini) [reply packages/kn-plugin-workflow/pkg/common/containers.go:58]: oh so this problem won't happen with podman?

Comment (treblereel) [reply packages/kn-plugin-workflow/pkg/common/containers.go:58]: @ricardozanini podman is awesome!

**Review by ricardozanini — APPROVED** [review]
@domhanak can you verify it locally?

**Comment by domhanak** [comment]
@ricardozanini looking

File: packages/kn-plugin-workflow/pkg/common/containers_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+		{"docker.io/apache/incubator-kie-sonataflow-devmode", []string{"docker.io/apache/incubator-kie-sonataflow-devmode:latest"}, true},
+		{"docker.io/apache/incubator-kie-sonataflow-devmode", []string{"apache/incubator-kie-sonataflow-devmode:latest"}, true},
+
+		{"apache/incubator-kie-sonataflow-devmode:main", []string{"docker.io/apache/incubator-kie-sonataflow-devmode:main"}, true},
+		{"apache/incubator-kie-sonataflow-devmode:main", []string{"apache/incubator-kie-sonataflow-devmode:main"}, true},
+
+		{"docker.io/apache/incubator-kie-sonataflow-devmode:main", []string{"incubator-kie-sonataflow-devmode:main"}, false},
+		{"docker.io/apache/incubator-kie-sonataflow-devmode", []string{"incubator-kie-sonataflow-devmode:latest"}, false},
+		{"apache/incubator-kie-sonataflow-devmode:main", []string{"incubator-kie-sonataflow-devmode:main"}, false},
```
Comment (tiagobento) [inline packages/kn-plugin-workflow/pkg/common/containers_test.go]: Are these fictitious, test-only names? If so, please don't use the same names we're using for real artifacts. People without context might think this is some sort of "indirect dependency" on these images.

**Review by ricardozanini — APPROVED** [review]

**Review by tiagobento — APPROVED** [review]

**Review by domhanak — APPROVED** [review]

**Pull request merged**